flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -546,7 +546,13 @@ public ExecutionMode getExecutionMode() {
 	 */
 	@PublicEvolving
 	public void setDefaultInputDependencyConstraint(InputDependencyConstraint inputDependencyConstraint) {
-		this.defaultInputDependencyConstraint = inputDependencyConstraint;
+		if (inputDependencyConstraint != null) {
+			this.defaultInputDependencyConstraint = inputDependencyConstraint;
+		} else {
+			// defaultInputDependencyConstraint is not allowed to be null
+			// setting it to ANY to not break existing jobs
+			this.defaultInputDependencyConstraint = InputDependencyConstraint.ANY;
+		}
 	}
 
```

Comment:
I think this change is trivial, to reduce code redundancy I did not add any test case for it.
I manually verified the change by invoking ExecutionConfig#setDefaultInputDependencyConstraint in BatchFineGrainedRecoveryITCase. 
Without this change there will be job submission error and the case will fail. 
With this change, the case can pass.